### ISSUE TITLE
TabsController - tabItem colors for states.

### ISSUE THREAD

**Issue by OrkhanAlikhanov** [CosmicMind/Material#860]
By the way, I was using the tabBar delegate to change title colors for (de)selected states accordingly. 
```swift
func tabBar(tabBar: TabBar, willSelect tabItem: TabItem) {
    tabBar.tabItems.forEach {
        $0.titleColor = .darkGray
    }    
    tabItem.titleColor = App.colors.primary
}
```
Shouldn't be there something like `selectedTabItemTitleColor` and `normalTabItemTitleColor`. ?
![image](https://user-images.githubusercontent.com/15037839/30227686-8ec2a766-94eb-11e7-9678-04f5cfa81066.png)

**Comment by danieldahan** [comment]
I can add this with the delegation calls.

**Comment by yasirmturk** [comment]
+100

**Comment by yasirmturk** [comment]
i do this on viewdidload

**Comment by danieldahan** [comment]
Please find this feature available in the latest [Material 2.12.0](https://github.com/CosmicMind/Material/releases/tag/2.12.0) release. 

Example: 

```swift
tabBar.setColor(Color.grey.base, for: .normal)
tabBar.setColor(Color.blue.base, for: .selected)
```

**Comment by markst** [comment]
@danieldahan I think `TabItemState` is unnecessary - why not just use the `UIControlState` enum?

Previously I would setup colours as follows:
```swift
override func prepare() {
    super.prepare()

    // Setup tab bar items appearance:
    self.viewControllers.forEach { (vc) in
        vc.tabItem.titleLabel?.font = UIFont.sparkyFont(.calibri, weight: .bold, size: 14)
        vc.tabItem.setTitleColor(UIColor.cs_Thunder(),for: .normal)
        vc.tabItem.setTitleColor(UIColor.cs_Citron(), for: .highlighted)
        vc.tabItem.setTitleColor(UIColor.cs_Citron(), for: .selected)
    }
}
```

However, now `updateColors()` overrides all of this & I don't have support for `.highlighted` state.

**Comment by markst** [comment]
https://github.com/CosmicMind/Material/issues/740

**Comment by danieldahan** [comment]
@markst yeah, we can make that adjustment.

**Comment by danieldahan** [comment]
Adding this suggestion https://github.com/CosmicMind/Material/issues/916#issuecomment-337786413

**Comment by danieldahan** [comment]
@markst I opted out of using `UIControlState` as it has more states than necessary for managing the tab items. That said, the latest interface for colors looks like this: 

```swift
tabBar.setLineColor(Color.orange.base, for: .selected)
        
tabBar.setTabItemsColor(Color.grey.base, for: .normal)
tabBar.setTabItemsColor(Color.purple.base, for: .selected)
tabBar.setTabItemsColor(Color.green.base, for: .highlighted)
```

This is available in [Material 2.12.2](https://github.com/CosmicMind/Material/releases/tag/2.12.2).

**Comment by OrkhanAlikhanov** [comment]
@danieldahan Thank you!
